# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ThirdPartyUtil.h
# Commit: fbf55a44d7fb
# Full Hash: fbf55a44d7fb72908eeb5d041b70ce2c503a2532
# Author: Anny Gakhokidze <agakhokidze@mozilla.com>
# Date: 2020-04-15 04:09:17
# Regressor Bug: 1594529
# File Overlap Count: 1
# Description:
#   Bug 1594529 - Create LoadInfo for subdocuments directly in parent process with DocumentChannel. r=mattwoodrow,nika
#   
#   Currently, with Fission enabled we are not able to create a proper LoadInfo
#   object when doing a subdocument load because we do not have access to a loading
#   context if the load is happening inside of an OOP frame. To solve this problem,
# ==============================================================================

diff -r 4ba9a230586a -r fbf55a44d7fb dom/base/ThirdPartyUtil.h
--- a/dom/base/ThirdPartyUtil.h	Mon Apr 13 22:41:37 2020 +0000
+++ b/dom/base/ThirdPartyUtil.h	Mon Apr 13 23:00:43 2020 +0000
@@ -14,6 +14,7 @@
 #include "nsEffectiveTLDService.h"
 #include "nsString.h"
 #include "nsPIDOMWindow.h"
+#include "mozilla/dom/WindowGlobalParent.h"
 
 class nsIURI;
 class nsPIDOMWindowOuter;
@@ -28,6 +29,9 @@
   static void Startup();
   static ThirdPartyUtil* GetInstance();
 
+  nsresult IsThirdPartyGlobal(mozilla::dom::WindowGlobalParent* aWindowGlobal,
+                              bool* aResult);
+
  private:
   ~ThirdPartyUtil();
 